cluster-state-service/canary/pkg/tests/get_instance.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/clienttest/client/operations"
+	"github.com/blox/blox/cluster-state-service/canary/pkg/tests/util"
+	"github.com/blox/blox/cluster-state-service/canary/pkg/wrappers"
+	"github.com/pkg/errors"
+)
+
+const (
+	getInstanceClusterNamePrefix = canaryClusterNamePrefix + "GetInstance_"
+)
+
+func GetInstance(ecsWrapper wrappers.ECSWrapper, ec2Wrapper wrappers.EC2Wrapper, cssWrapper wrappers.CSSWrapper) error {
```

Comment:
would it be cleaner to have a struct with all the wrappers, an interface with all test methods (e.g. GetInstance, ListInstances, etc) and then implement the methods in separate files just like you have (so list_instances.go implements ListInstances) and only pass in parameters that that particular test needs e.g. clusterName

so for example

type canary struct {
  ecsWrapper
  cssWrapper
}

type Canary interface {
  getInstance(clusterName)
  listInstances()
}

func (c canary) getInstance(clusterName) {}

up to you, the way you have it is fine too.